src/pages/patients/Patients.tsx

```diff
@@ -45,14 +46,47 @@ const Patients = (props: PatientsProps) => {
         disabled: false,
     });
 
+    
+    const [getPatientsByStatusMutation] = useGetPatientsByStatusMutation();
+    
+    const handleGetPatientsByStatus = async () => {
+        try {
+            setIsLoading(true);
+            const statusList = Object.entries(filterStatus)
+                .filter(([key, value]) => value)
+                .map(([key]) => key);
+    
+            const data = await Promise.all(
+                statusList.map(async (status) => {
+                    const result = await getPatientsByStatusMutation({ status });
+                    return result as { data: Patient[] };
```

Comment:
just a question is this cast necessary? if in the service you already set the type of data you don't need to add it again here